### PR TITLE
Restore last case list query

### PR DESCRIPTION
**Pull request by phillipm**
cross-request: https://github.com/dimagi/commcare-odk/pull/795

### CODE REVIEW

File: tests/test/org/commcare/backend/session/test/SessionNavigatorTests.java [hunk 1]
```diff
@@ -144,5 +144,39 @@ public void testAutoSelectEnabledWithOneCaseAndNoDetail() {
         triggerSessionStepAndCheckResultCode(SessionNavigator.START_FORM_ENTRY);
     }
 
+    @Test
+    public void testSettingAndGettingSessionExtras() {
```
Comment (phillipm) [inline tests/test/org/commcare/backend/session/test/SessionNavigatorTests.java]: This is a pretty standard session traversal. Let me know if there are corner cases that I should add to the test.
